tornado/test/ioloop_test.py

```diff
@@ -584,6 +585,39 @@ def handle_exception(typ, value, traceback):
         self.assertEqual(self.exception.args[0], "callback")
         self.assertEqual(self.future.exception().args[0], "worker")
 
+    def test_run_in_executor_gen(self):
+        def func(arg1, arg2):
+            return arg1 + arg2
+
+        @gen.coroutine
+        def main():
+            arg1, arg2 = random.random(), random.random()
```

Comment:
Why introduce randomness? Also, comparing floats for equality is not a great idea. I think a better test would be to submit two functions to the executor and make sure that they run in parallel. 

    event1 = threading.Event()
    event2 = threading.Event()
    def callback(self_event, other_event):
        self_event.set()
        other_event.wait()
        return self_event
    res = yield [
        IOLoop.current().run_in_executor(None, callback, event1, event2),        
        IOLoop.current().run_in_executor(None, callback, event2, event1),
    ]
    self.assertEqual([event1, event2], res)